dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java

```diff
@@ -591,30 +642,150 @@ public void Test_Required_Params() throws IOException, DotDataException, DotSecu
             Assert.assertEquals(HttpStatus.SC_OK, appResponse.getStatus());
 
+            //Secrets are destroyed for security every time. Making the form useless. They need to be re-generated every time.
             //Here's a secret with an empty param that is marked as required.
-            final Map<String, Param> secretParam = ImmutableMap.of(
-                    "param1", Param.newParam("", false, Type.STRING, null, null, true)
+            final Map<String, Input> inputParamMap = ImmutableMap.of(
+                    "param1", Input.newInputParam("".toCharArray(),false));
+
+            final Host host = new SiteDataGen().nextPersisted();
+            final SecretForm secretForm = new SecretForm(inputParamMap);
+            final Response createSecretResponse = appsResource
+                    .createAppSecrets(request, response, key, host.getIdentifier(), secretForm);
+            Assert.assertEquals(HttpStatus.SC_BAD_REQUEST, createSecretResponse.getStatus());
+        }catch (Exception e){
+            Logger.error(AppsResourceTest.class, e);
+            fail();
+        }
+    }
+
+    @Test
+    public void Test_Required_Params_Multiple_Params_Descriptor_Non_Empty_Value_Missing_Required_Param_Sent()  {
+
+        final SortedMap<String, ParamDescriptor> initialParamsMap = ImmutableSortedMap.of(
+                "param1", ParamDescriptor.newParam("val-1", false, Type.STRING, "label", "hint", true),
+                "param2", ParamDescriptor.newParam("val-1", false, Type.STRING, "label", "hint", true)
+        );
+
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpServletResponse response = mock(HttpServletResponse.class);
+
+        when(request.getRequestURI()).thenReturn("/baseURL");
+
+        long time = System.currentTimeMillis();
+
+        final String key = String.format("all_lower_case_not_too_short_prefix_%d", time);
+        final String fileName = String.format("%s.yml", key);
+
+        //We're indicating that extra params are allowed to test required params are still required
+        try(final InputStream inputStream = createAppDescriptorFile(fileName, key,
+                key,
+                "Test-required-params",
+                true, initialParamsMap)){
+
+            final Response appResponse = appsResource
+                    .createApp(request, response,
+                            createFormDataMultiPart(fileName, inputStream));
+            Assert.assertNotNull(appResponse);
+            Assert.assertEquals(HttpStatus.SC_OK, appResponse.getStatus());
+
+            //Secrets are destroyed for security every time. Making the form useless. They need to be re-generated every time.
+            //We're sending only one parameter when the descriptor says there's another one mandatory.
+            final Map<String, Input> inputParamMap = ImmutableMap.of(
+                    "param1", Input.newInputParam("any-value".toCharArray(),false)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "any-value" appears 5 times in this file; the first occurrence is on line 685](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5289044)